### PR TITLE
A4A: Fix missing referral external links.

### PR DESCRIPTION
**Pull request by jkguidaven**
Closes https://github.com/Automattic/jetpack-genesis/issues/381

## Proposed Changes

* Remove the 'How does it work?' link on the Dashboard page.

| Before | After |
|--------|--------|
| Screenshot 2024-06-18 at 6 56 04 PM | Screenshot 2024-06-18 at 6 55 38 PM |

* Redirect Request payment checkouts learn more link to  https://agencieshelp.automattic.com/knowledge-base/billing-and-payments/
Screenshot 2024-06-18 at 7 01 37 PM

* Redirect the user to https://automattic.com/for-agencies/platform-agreement/ when clicking the TOS link on the Client's checkout page.
*
Screenshot 2024-06-18 at 8 24 29 PM

## Testing Instructions

* Use the A4A live link and go to `/referrals/dashboard`
* Confirm that the 'How does it work?' link is no longer visible. **If you already have some referrals, you may need to create a new agency account to access the empty dashboard.**
* Go to `/marketplace/products`
* Create some referrals by toggling 'refer products' and adding items to the cart.
* Checkout the items.
* Confirm that the learn more link redirects to the Billing and payment KB.
* Proceed to checkout with your client account.
* In the client checkout form, confirm that the TOS link works.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by yashwin — CHANGES_REQUESTED** [review]
Thanks for working on this, @jkguidaven!

Could we please update the link [here](https://github.com/Automattic/wp-calypso/blob/5f3c7b4d5862af8ddda93e5dbe1c463010e61d78/client/a8c-for-agencies/sections/marketplace/checkout/submit-payment-info.tsx#L27-L28)?

**Comment by jkguidaven** [comment]
@yashwin good catch this has been addressed.

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comment, @jkguidaven!

LGTM!

Let's wait for @madebynoam approval here.

**Pull request merged**
